src/bindgen/ir/generic_path.rs

```diff
@@ -274,6 +320,10 @@ impl GenericPath {
         let last_segment = path.segments.last().unwrap();
         let name = last_segment.ident.unraw().to_string();
 
+        let assoc: Option<AssocTypeId> = if let Ok(id) = AssocTypeId::new(path, qself){
```

Comment:
```suggestion
        let assoc = if let Ok(id) = AssocTypeId::new(path, qself){
```

type inference should handle it